src/common/oscap_acquire.c

```diff
@@ -326,18 +326,59 @@ char* oscap_acquire_url_download(const char *url, size_t* memory_size)
 		return NULL;
 	}
 
-	struct oscap_buffer* buffer = oscap_buffer_new();
-
-	curl_easy_setopt(curl, CURLOPT_URL, url);
-	curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, write_to_memory_callback);
-	curl_easy_setopt(curl, CURLOPT_WRITEDATA, buffer);
-	curl_easy_setopt(curl, CURLOPT_ACCEPT_ENCODING, "");
-	curl_easy_setopt(curl, CURLOPT_TRANSFER_ENCODING, true);
-	curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, true);
-	curl_easy_setopt(curl, CURLOPT_VERBOSE, true);
-	curl_easy_setopt(curl, CURLOPT_DEBUGFUNCTION, _curl_trace);
-
-	CURLcode res = curl_easy_perform(curl);
+	CURLcode res;
+
+	res = curl_easy_setopt(curl, CURLOPT_URL, url);
+	if (res != 0) {
```

Comment:
We should also add `curl_easy_cleanup(curl)` in these newly added ifs to free the allocated structure by the `curl_easy_init` function.